### PR TITLE
Lostsomething

### PR DESCRIPTION
**Pull request by JannikUekoetter**
I have written a form that allows you to add new lost items. 
Below the form the original array of dummy data is shown and on submit of the form a new item is added.
Currently a photo of Mr Bean is used for the new item, because the image upload is not working yet. For the working image upload a workshop will be held. For this reason, Mr Bean is currently the workaround, so that the representation can be tested. This will be edited after completion of the workshop with the knowledge from it.

### CODE REVIEW

File: components/AddLostItem.js [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+import { nanoid } from "nanoid";
+import Image from "next/image";
+
+export default function AddLostItem() {
+  const [description, setDescription] = useState("");
+  const [location, setLocation] = useState("");
+  const [name, setName] = useState("");
+  const [lost, setLost] = useState(lostItems);
+  const [img, setImage] = useState("/assets/MrBean.png");
+
```
Comment (CathrinTruchan) [inline components/AddLostItem.js:14]: Hey Jannik, I think you can make it much easier for yourself by using just one state and update the different properties in the object. Or do you need these different states for something else? Then you don't need the onChange-functions and catch all the data when the form is submitted.

File: components/AddLostItem.js [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+          <h2>Tell us more</h2>
+          <label htmlFor="name">
+            <input
+              key="name"
+              type="text"
+              placeholder="Name"
+              required
+              value={name}
+              onChange={(event) => setName(event.target.value)}
+            />
```
Comment (anhthu253) [inline components/AddLostItem.js:36]: <input
              name="name"
              type="text"
              placeholder="Name"
              required
            />

File: components/AddLostItem.js [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+          <br />
+          <h3>Whats your location?</h3>
+          <input
+            key="location"
+            type="text"
+            placeholder="Location"
+            required
+            value={location}
+            onChange={(event) => setLocation(event.target.value)}
+          />
```
Comment (anhthu253) [inline components/AddLostItem.js:63]: <input
            name="location"
            type="text"
            placeholder="Location"
            required
          />

File: components/AddLostItem.js [hunk 4]
```diff
@@ -0,0 +1,127 @@
...
+            type="text"
+            key="description"
+            htmlFor="description"
+            placeholder="Description"
+            rows="15"
+            cols="35"
+            required
+            value={description}
+            onChange={(event) => setDescription(event.target.value)}
+          />
```
Comment (anhthu253) [inline components/AddLostItem.js:50]: <StyledTextarea
            type="text"
            name="description"
            placeholder="Description"
            rows="15"
            cols="35"
            required
          />

File: components/AddLostItem.js [hunk 5]
```diff
@@ -0,0 +1,127 @@
...
+        <label htmlFor="picture">
+          <h4>Upload a Picture</h4>
+        </label>
+        <input
+          key="picture"
+          type="file"
+          id="picture"
+          name="picture"
+          accept="image/png, image/jpeg"
+        />
```
Comment (anhthu253) [inline components/AddLostItem.js:76]: <input
          name="picture"
          type="file"
          id="picture"
          name="picture"
          accept="image/png, image/jpeg"
        />

File: components/AddLostItem.js [hunk 6]
```diff
@@ -0,0 +1,127 @@
...
+  const [description, setDescription] = useState("");
+  const [location, setLocation] = useState("");
+  const [name, setName] = useState("");
+  const [lost, setLost] = useState(lostItems);
+  const [img, setImage] = useState("/assets/MrBean.png");
+
+  const randomId = nanoid();
+
+  const handleClick = (event) => {
+    event.preventDefault();
```
Comment (anhthu253) [inline components/AddLostItem.js:18]: const formdata = new FormData(event.target);
    const data = Object.fromEntries(formdata);
const newLostItem = {...data,id: Math.random().toString(36).substring(2)}

File: components/AddLostItem.js [hunk 7]
```diff
@@ -0,0 +1,127 @@
...
+    const newLostItem = { name, description, location, img, randomId };
```
Comment (anhthu253) [inline components/AddLostItem.js:19]: const newLostItem = { name, description, location, img, randomId } -> you dont need this anymore

File: components/AddLostItem.js [hunk 8]
```diff
@@ -0,0 +1,127 @@
...
+        <label>
+          <h2>Tell us more</h2>
+          <label htmlFor="name">
+            <input
+              key="name"
+              type="text"
+              placeholder="Name"
+              required
+              value={name}
+              onChange={(event) => setName(event.target.value)}
```
Comment (anhthu253) [inline components/AddLostItem.js:35]: name="name"
type="text"
placeholder="Name"
required

File: components/AddLostItem.js [hunk 9]
```diff
@@ -0,0 +1,127 @@
...
+          <StyledTextarea
+            type="text"
+            key="description"
+            htmlFor="description"
+            placeholder="Description"
+            rows="15"
+            cols="35"
+            required
+            value={description}
+            onChange={(event) => setDescription(event.target.value)}
```
Comment (anhthu253) [inline components/AddLostItem.js:49]: type="text"
name="description"
placeholder="Description"
rows="15"
cols="35"
required

File: components/AddLostItem.js [hunk 10]
```diff
@@ -0,0 +1,127 @@
...
+        <br />
+        <label htmlFor="picture">
+          <h4>Upload a Picture</h4>
+        </label>
+        <input
+          key="picture"
+          type="file"
+          id="picture"
+          name="picture"
+          accept="image/png, image/jpeg"
```
Comment (anhthu253) [inline components/AddLostItem.js:75]: name="picture"
          type="file"
          id="picture"
          name="picture"
          accept="image/png, image/jpeg"

File: components/AddLostItem.js [hunk 11]
```diff
@@ -0,0 +1,127 @@
...
+        <br />
+        <label htmlFor="location">
+          <br />
+          <h3>Whats your location?</h3>
+          <input
+            key="location"
+            type="text"
+            placeholder="Location"
+            required
+            value={location}
```
Comment (anhthu253) [inline components/AddLostItem.js:61]: name="location"
type="text"
placeholder="Location"
required

File: components/AddLostItem.js (hunk 7 above)
Comment (anhthu253) [inline components/AddLostItem.js:19]: const formdata = new FormData(event.target);
const data = Object.fromEntries(formdata);
const newLostItem = {...data,id: Math.random().toString(36).substring(2)}

File: components/AddLostItem.js [hunk 12]
```diff
@@ -0,0 +1,127 @@
...
+import Image from "next/image";
+
+export default function AddLostItem() {
+  const [description, setDescription] = useState("");
+  const [location, setLocation] = useState("");
+  const [name, setName] = useState("");
+  const [lost, setLost] = useState(lostItems);
+  const [img, setImage] = useState("/assets/MrBean.png");
+
+  const randomId = nanoid();
```
Comment (anhthu253) [inline components/AddLostItem.js:15]: u dont need this anymore

File: components/AddLostItem.js (hunk 7 above)
Comment (JannikUekoetter) [reply components/AddLostItem.js:19]: @anhthu253 where will the state be used in this example? I think the " const [lost, setLost] = useState(lostItems);" hast to remain in my code to use this data on another page (after lifting the state).

File: components/AddLostItem.js [hunk 13]
```diff
@@ -0,0 +1,107 @@
...
+          onChange={setLost.img}
+        />
+        <br />
+        <br />
+        <br />
+        <button type="submit">Submit</button>
+      </StyledForm>
+      <StyledNewItems>
+        {lost.map((setLost) => {
+          return (
```
Comment (CathrinTruchan) [inline components/AddLostItem.js:65]: Hi Jannik, you named the variable setLost like the setter-function of your state. That was a bit confusing for me. Maybe you can find a better name ;)

**Review by CathrinTruchan — APPROVED** [review]
Looks good to me now 🥳
It just left a short comment :)

**Pull request merged**